src/PrestaShopBundle/Controller/Admin/Sell/Customer/CustomerController.php

```diff
@@ -441,7 +441,7 @@ public function setRequiredFieldsAction(Request $request)
     public function searchAction(Request $request)
     {
         $query = $request->query->get('customer_search');
-        $phrases = explode(' ', $query);
+        $phrases = explode('OR', $query);
```

Comment:
@okom3pom Thank you for this great proposal.

Perhaps the implementation requires further testing to avoid unexpected results:
1. The OR separator might need space characters around `' OR '`, in order to handle some tricky cases like `"SPECTOR Warren"`.
2. Beyond this first splitting, if I'm not wrong, `Customer::searchByName()` will also split each individual search query with space separator: https://github.com/PrestaShop/PrestaShop/blob/cf54f94ec6440e573a1f7b1549ee29ce31e619c3/classes/Customer.php#L875

Reply:
@saulaski 
 _The OR separator might need space characters around ' OR ', in order to handle some tricky cases like "SPECTOR Warren"._
 
 Nice catch I didn't think of that, for the second point i need to check

Reply:
For the second point is for this.

https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/Customer/QueryHandler/SearchCustomersHandler.php#L83

`$search_items = explode(' ', $query); ` is for search **Thomas Roux** in firstname and lastname

Reply:
@okom3pom Thank you for this quick validation